### PR TITLE
feat(shorebird_cli): upgrade to latest shorebird (3.7.10)

### PR DESCRIPTION
**Pull request by felangel**
## Description

- feat(shorebird_cli): upgrade to latest shorebird (3.7.10)

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Review by eseidel — APPROVED** [review]
lgtm

**Review by eseidel — APPROVED** [review]

File: packages/shorebird_cli/lib/src/cache.dart [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    await Isolate.run(
+      () async {
+        final inputStream = InputFileStream(archivePath);
+        final archive = ZipDecoder().decodeBuffer(inputStream);
+        cache.extractArchive(archive, outputPath);
+      },
+    );
+
+    for (final executable in executables) {
+      final process = await Process.start(
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/cache.dart:100]: I suspect this won't work on windows, but that can also be a next-week problem.

File: packages/shorebird_cli/lib/src/cache.dart [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  final Platform platform;
+
+  @override
+  String get name => 'patch';
+
+  @override
+  List<String> get executables => ['patch'];
+
+  @override
+  String get storagePath {
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/cache.dart:119]: This ends up abstracted, but makes sense on our single artifact for now. 🤣

**Review by eseidel — APPROVED** [review]

File: packages/shorebird_cli/lib/src/cache.dart (hunk 2 above)
Comment (felangel) [reply packages/shorebird_cli/lib/src/cache.dart:119]: yeah a lot of this needs to be abstracted more/differently

File: packages/shorebird_cli/lib/src/cache.dart (hunk 1 above)
Comment (felangel) [reply packages/shorebird_cli/lib/src/cache.dart:100]: yeah good catch

Comment (felangel) [reply packages/shorebird_cli/lib/src/cache.dart:100]: fixed 👍

**Review by eseidel — APPROVED** [review]
lgtm

**Pull request merged**